BlueprintUI/Sources/Layout/Layout.swift

```diff
@@ -1,29 +1,125 @@
-import UIKit
+import CoreGraphics
 
-/// Conforming types can calculate layout attributes for an array of children.
-public protocol Layout {
+/// A type that defines the geometry of a collection of elements.
+///
+/// You traditionally arrange views in your app's user interface using built-in layout containers
+/// like ``Row`` and ``Column``. If  you need more complex layout behavior, you can define a custom
+/// layout container by creating a type that conforms to the ``Layout`` protocol and implementing
+/// its required methods:
+///
+/// - ``CaffeinatedLayout/sizeThatFits(proposal:subelements:cache:)`` reports the sizes of the
+///   composite layout.
+/// - ``CaffeinatedLayout/placeSubelements(in:subelements:cache:)`` assigns positions to the
+///   container's subelements.
+///
+/// You can define a basic layout type with only these two methods (see note below):
+///
+/// ```swift
+/// struct BasicLayout: Layout {
+///     func sizeThatFits(
+///         proposal: SizeConstraint,
+///         subelements: Subelements,
+///         cache: inout ()
+///     ) -> CGSize {
+///         // Calculate and return the size of the layout container.
+///     }
+///
+///     func placeSubelements(
+///         in size: CGSize,
+///         subelements: Subelements,
+///         cache: inout ()
+///     ) {
+///         // Tell each subelement where to appear.
+///     }
+/// }
+/// ```
+///
+/// Add your layout to an element by passing it to the `ElementContent` initializer
+/// ``ElementContent/init(layout:configure:)``. If your layout has parameters that come from the
+/// element, pass them at initialization time. Use the `configure` block to add your element's
+/// children to the content.
+///
+/// ```swift
+/// struct BasicContainer: Element {
+///     var alignment: Alignment = .center
+///     var children: [Element]
+///
+///     var content: ElementContent {
+///         ElementContent(layout: BasicLayout(alignment: alignment)) { builder in
+///             for child in children {
+///                 builder.add(element: child)
+///             }
+///         }
+///     }
+///
+///     func backingViewDescription(with context: ViewDescriptionContext) -> ViewDescription? {
+///         nil
+///     }
+/// }
+/// ```
+///
+/// If your layout is specialized for laying out a single subelement, you can use the
+/// ``SingleChildLayout`` protocol instead. It has similar methods, but is strongly typed for a
+/// single subelement instead of a collection.
+///
+/// - Note: During the transition from Blueprint's legacy layout system to Caffeinated Layout, the
+///   ``Layout`` protocol is composed of two sets of layout methods: ``LegacyLayout`` and
+///   ``CaffeinatedLayout``. While this documentation focuses on the newer layout API, you must
+///   currently implement both. Fortunately, the methods are similar, and you may be able to reuse
+///   logic.
```

Comment:
Good callout. Will we also need to advise on how to toggle between the layout systems for testing purposes?

Reply:
> Good callout. Will we also need to advise on how to toggle between the layout systems for testing purposes?

Like, unit and snapshot testing, or just for experimentation? Currently the advice on `LayoutMode` is to stick with the default mode. Normally I wouldn't expect folks who are writing elements to need to explicitly test both modes.